src/Mvc/Mvc.Core/src/JsonOptions.cs

```diff
@@ -11,6 +12,21 @@ namespace Microsoft.AspNetCore.Mvc
     /// </summary>
     public class JsonOptions
     {
+        /// <summary>
+        /// Gets or sets a flag to determine whether error messages from JSON deserialization by the
+        /// <see cref="SystemTextJsonInputFormatter"/> will be added to the <see cref="ModelStateDictionary"/>. If
+        /// <see langword="false"/>, a generic error message will be used instead.
+        /// </summary>
+        /// <value>
+        /// The default value is <see langword="true"/>.
+        /// </value>
+        /// <remarks>
+        /// Error messages in the <see cref="ModelStateDictionary"/> are often communicated to clients, either in HTML
+        /// or using <see cref="BadRequestObjectResult"/>. In effect, this setting controls whether clients can receive
+        /// detailed error messages about submitted JSON data.
+        /// </remarks>
+        public bool AllowSystemTextJsonInputFormatterExceptionMessages { get; set; } = true;
```

Comment:
If we are adding this, I think we should have a behavior change here and make it false by default. @pranavkm thoughts?

Do we have this in general as a feature? If not, we might consider having this as a general modelbinding flag the same way we do in Blazor Server

Reply:
We have an equivalent option for JSON.NET, but not one for System.Text.Json. Adding this at the top level is a bit clunky - XML would be unable to use it because the exceptions produced by the formatters are not considered safe to presented to the client. 

The one choice we could consider is to invert this option `SuppressSystemTextJsonInputFormatterExceptionMessages`. It does make it a little less consistent with https://docs.microsoft.com/en-us/dotnet/api/microsoft.aspnetcore.mvc.mvcnewtonsoftjsonoptions.allowinputformatterexceptionmessages?view=aspnetcore-5.0#Microsoft_AspNetCore_Mvc_MvcNewtonsoftJsonOptions_AllowInputFormatterExceptionMessages, but more consistent with the rest of our API design. I can bring this up during api review.

Reply:
```suggestion
        public bool AllowInputFormatterExceptionMessages { get; set; } = true;
```

Reply:
I renamed this to `AllowInputFormatterExceptionMessages`, let me know if I need to do anything else.